<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ONDA - Basculeur</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* --- Contenu Principal --- */
#mainContent {
	background-color: #FFFFFF;
	padding: 1px;
}

#titleLabel {
	font-family: Source Sans Pro; 
	font-size: 30px;
	font-weight: 700;
	color: #171A1F;
}

#mainContentFrame {
	background-color: #FFFFFF;
	border: None;
	border-radius: 0px;
}

/* --- Cartes --- */
.QFrame {
	background-color: #FFFFFF;
	border-radius: 24px;
	border: 1px solid #E5E7EB;
}

.cardTitleLabel {
	font-family: Source Sans Pro;
	font-size: 20px;
	font-weight: 700;
	color: #171A1F;
	padding-bottom: 5px;
}


</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="mainContentFrame">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,1">
       <property name="spacing">
        <number>24</number>
       </property>
       <property name="leftMargin">
        <number>32</number>
       </property>
       <property name="topMargin">
        <number>24</number>
       </property>
       <property name="rightMargin">
        <number>32</number>
       </property>
       <property name="bottomMargin">
        <number>24</number>
       </property>
       <item>
        <widget class="QFrame" name="headerFrame">
         <property name="styleSheet">
          <string notr="true">border: none;</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="titleLabel">
            <property name="font">
             <font>
              <family>Source Sans Pro</family>
              <pointsize>-1</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Basculeur</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="alertRed">
         <property name="styleSheet">
          <string notr="true">#alertRed {
	width: 1120px; 
	height: 74px; 
	background-color: #991B1B;
	border-radius: 10px;
	border-width: 1px; 
	border-color: #991B1B; 
	border-style: solid; 
}</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="iconAlertRed">
            <property name="minimumSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: #991B1B;</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="pixmap">
             <pixmap resource="../assets/icons.qrc">:/icons/icons/exclamation-circle.svg</pixmap>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignHCenter|Qt::AlignmentFlag::AlignTop</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLabel" name="alertRedTitle">
              <property name="styleSheet">
               <string notr="true">background-color: #991B1B;
font-family: Open Sans; 
font-size: 16px; 
line-height: 16px; 
font-weight: 500; 
color: #FFFFFF;</string>
              </property>
              <property name="text">
               <string>Défaillance canal principal</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="alertRedBody">
              <property name="styleSheet">
               <string notr="true">background-color: #991B1B;
color: #FFFFFF;
font-weight: 600;
font-size: 14px;
font-family: Roboto; </string>
              </property>
              <property name="text">
               <string>Une défaillance a été détectée sur le canal principal. Le basculement automatique est recommandé.</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="statusFrame">
         <property name="styleSheet">
          <string notr="true">border-radius: 12px;</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>13</number>
          </property>
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="topMargin">
           <number>20</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <property name="bottomMargin">
           <number>20</number>
          </property>
          <item>
           <widget class="QLabel" name="statusTitleLabel">
            <property name="font">
             <font>
              <family>Source Sans Pro</family>
              <pointsize>-1</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>État de l'équipement</string>
            </property>
            <property name="class" stdset="0">
             <string>cardTitleLabel</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_2">
            <property name="styleSheet">
             <string notr="true">border: none;</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::Shape::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Shadow::Plain</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="spacing">
              <number>8</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label">
               <property name="styleSheet">
                <string notr="true">	font-family: Open Sans; 
	font-size: 20px; 
	line-height: 28px; 
	font-weight: 700; 
	color: #1570D1; </string>
               </property>
               <property name="text">
                <string>Équipement actif :</string>
               </property>
               <property name="indent">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="font">
                <font>
                 <family>Open Sans</family>
                 <pointsize>-1</pointsize>
                 <bold>true</bold>
                </font>
               </property>
               <property name="styleSheet">
                <string notr="true">	font-family: Open Sans; 
	font-size: 20px; 
	line-height: 28px; 
	font-weight: 700; 
	color: #1570D1; </string>
               </property>
               <property name="text">
                <string>Principal</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="statusBadge">
               <property name="styleSheet">
                <string notr="true">background-color: #DCFCE7;
color: #166534;
border-radius: 8px;
padding: 2px 8px;
font-weight: 500;</string>
               </property>
               <property name="text">
                <string>Actif</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame_3">
         <property name="styleSheet">
          <string notr="true">border: none;</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="spacing">
           <number>15</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="btnManuel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">/* Button */
QPushButton {
	width: 130.78125px; 
	height: 40px; 
	padding: 0 12px; 
	font-family: Roboto; 
	font-size: 14px; 
	line-height: 22px; 
	font-weight: 500; 
	color: #FFFFFF; 
	background: #1570D1; 
	opacity: 1; 
	border: none; 
	border-radius: 6px; 
}

/* Hover */
QPushButton:hover {
	background-color: #0E4A8A;
}

/* Pressed */
QPushButton:hover:active {
  background: #09305A; 
}
</string>
            </property>
            <property name="text">
             <string>Basculement manuel</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnConfigurer">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">/* Button */
QPushButton {
	width: 220.75px; 
	height: 38.45px; 
	padding: 0 12px;  
	font-family: Roboto; 
	font-size: 14px; 
	line-height: 22px; 
	font-weight: 500; 
	color: #1570D1; 
	background: #FFFFFF; 
	opacity: 1; 
	border-radius: 6px; 
	border-width: 1px; 
	border-color: #1570D1; 
	border-style: solid; 
}

/* Hover */
QPushButton:hover {
	background-color: #374151; 
}

/* Pressed */
QPushButton:hover:active {
	background-color: #F9FAFB;
}
</string>
            </property>
            <property name="text">
             <string>Configurer bascule automatique</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="historyFrame">
         <property name="styleSheet">
          <string notr="true">border: none;</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Plain</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <property name="spacing">
           <number>12</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="historyTitleLabel">
            <property name="font">
             <font>
              <family>Source Sans Pro</family>
              <pointsize>-1</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Historique des basculements</string>
            </property>
            <property name="class" stdset="0">
             <string>cardTitleLabel</string>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignmentFlag::AlignHCenter|Qt::AlignmentFlag::AlignVCenter">
           <widget class="QTableWidget" name="tableWidget">
            <property name="minimumSize">
             <size>
              <width>800</width>
              <height>0</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">/* Table */
QTableWidget {
	border: 1px solid #E5E7EB;
	border-radius: 6px;
	width: 1123px; 
	height: 242.5px;
	gridline-color: #E5E7EB;
	opacity: 1; 
}

QHeaderView::section {
	font-family: Roboto; 
	background-color: #F9FAFB;
	color: #565D6D;
	font-size: 14px;
	padding: 8px;
	border: none;
	border-bottom: 1px solid #E5E7EB;
	font-weight: 500;
}

QTableWidget::item {
	font-family: Roboto; 
	font-size: 14px; 
	padding-left: 8px;
	font-weight: 400;
	color: #171A1F;
}
</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::Shape::StyledPanel</enum>
            </property>
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
            <property name="autoScroll">
             <bool>true</bool>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectionBehavior::SelectItems</enum>
            </property>
            <property name="horizontalScrollMode">
             <enum>QAbstractItemView::ScrollMode::ScrollPerItem</enum>
            </property>
            <property name="showGrid">
             <bool>true</bool>
            </property>
            <property name="gridStyle">
             <enum>Qt::PenStyle::SolidLine</enum>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <attribute name="horizontalHeaderVisible">
             <bool>true</bool>
            </attribute>
            <attribute name="horizontalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>130</number>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <column>
             <property name="text">
              <string>Date/Heure</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Type d'événement</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Description</string>
             </property>
            </column>
            <item row="0" column="0">
             <property name="text">
              <string>2023-10-26 14:30</string>
             </property>
            </item>
            <item row="0" column="1">
             <property name="text">
              <string>Manuel</string>
             </property>
            </item>
            <item row="0" column="2">
             <property name="text">
              <string>Principal</string>
             </property>
            </item>
            <item row="1" column="0">
             <property name="text">
              <string>2023-10-25 09:15</string>
             </property>
            </item>
            <item row="1" column="1">
             <property name="text">
              <string>Défaillance</string>
             </property>
            </item>
            <item row="1" column="2">
             <property name="text">
              <string>Principal</string>
             </property>
            </item>
            <item row="2" column="0">
             <property name="text">
              <string>2023-10-24 18:00</string>
             </property>
            </item>
            <item row="2" column="1">
             <property name="text">
              <string>Automatique</string>
             </property>
            </item>
            <item row="2" column="2">
             <property name="text">
              <string>Secours</string>
             </property>
            </item>
            <item row="3" column="0">
             <property name="text">
              <string>2023-10-23 11:45</string>
             </property>
            </item>
            <item row="3" column="1">
             <property name="text">
              <string>Manuel</string>
             </property>
            </item>
            <item row="3" column="2">
             <property name="text">
              <string>Principal</string>
             </property>
            </item>
            <item row="4" column="0">
             <property name="text">
              <string>2023-10-22 07:00</string>
             </property>
            </item>
            <item row="4" column="1">
             <property name="text">
              <string>Défaillance</string>
             </property>
            </item>
            <item row="4" column="2">
             <property name="text">
              <string>Secours</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="../assets/icons.qrc"/>
 </resources>
 <connections/>
</ui>
